### PR TITLE
Add event onClientPedStep

### PR DESCRIPTION
**Pull request by CrosRoad95**
Type: **Clientside**
Name: **onClientPedStep**
Arguments: `bool foot`
if foot is equal `true` then ped put his left foot, otherwise he put right foot
Usage: Mainaly for sound system, you can detect which material hit foot and play different sounds for grass, sand, rocks etc.
LosFaul also suggested that will be able to make fitness step counter.

Test resource: ( you have to rename events to `onClientPedStep` )
[onPedTakeAStep.zip](https://github.com/multitheftauto/mtasa-blue/files/2148960/onPedTakeAStep.zip)

Player
event test
Ped
event test for ped

### CODE REVIEW

**Comment by qaisjp** [comment]
Hey - this looks pretty interesting! Well done.

My first comment for improvement would be to merge it into one event: `onClientPedStep`

File: Client/mods/deathmatch/logic/CClientGame.cpp [hunk 1]
```diff
@@ -454,6 +455,7 @@ CClientGame::~CClientGame(void)
     pKeyBinds->SetAllControlsEnabled(true, true, true);
     g_pCore->ForceCursorVisible(false);
     SetCursorEventsEnabled(false);
+    g_pMultiplayer->SetTakePedAStepHandler(NULL);
```
Comment (botder) [inline Client/mods/deathmatch/logic/CClientGame.cpp]: Please use **nullptr** and move the line up to the other `g_pMultiplayer->...` lines

File: Client/mods/deathmatch/logic/CClientGame.cpp [hunk 2]
```diff
@@ -2776,6 +2778,7 @@ void CClientGame::AddBuiltInEvents(void)
     m_Events.AddEvent("onClientPedChoke", "", NULL, false);
     m_Events.AddEvent("onClientPedHeliKilled", "heli", NULL, false);
     m_Events.AddEvent("onClientPedHitByWaterCannon", "vehicle", NULL, false);
+    m_Events.AddEvent("onClientPedStep", "foot", NULL, false);
```
Comment (botder) [inline Client/mods/deathmatch/logic/CClientGame.cpp]: Use **nullptr** instead of NULL.

File: Client/multiplayer_sa/CMultiplayerSA.cpp [hunk 3]
```diff
@@ -358,6 +358,7 @@ ObjectDamageHandler*        m_pObjectDamageHandler = NULL;
 ObjectBreakHandler*         m_pObjectBreakHandler = NULL;
 FxSystemDestructionHandler* m_pFxSystemDestructionHandler = NULL;
 DrivebyAnimationHandler*    m_pDrivebyAnimationHandler = NULL;
+TakeAStepHandler*           m_pTakeAStepHandler = NULL;
```
Comment (botder) [inline Client/multiplayer_sa/CMultiplayerSA.cpp]: Use **nullptr** here.

File: Client/multiplayer_sa/multiplayer_shotsync.cpp [hunk 4]
```diff
@@ -71,18 +71,23 @@ extern DeathHandler*          m_pDeathHandler;
 extern FireHandler*           m_pFireHandler;
 extern ProjectileHandler*     m_pProjectileHandler;
 extern ProjectileStopHandler* m_pProjectileStopHandler;
+extern TakeAStepHandler*      m_pTakeAStepHandler;
 
 char szDebug[255] = {'\0'};
 
 DWORD   RETURN_CProjectile__AddProjectile = 0x401C3D;
 DWORD   RETURN_CProjectile__CProjectile = 0x4037B3;
+DWORD   CONTINUE_CPed_DoFootLanded = 0x5E5386;
```
Comment (botder) [inline Client/multiplayer_sa/multiplayer_shotsync.cpp]: Prefix your variable with `static const` to keep the value in this translation unit and constant.

File: Client/multiplayer_sa/multiplayer_shotsync.cpp [hunk 5]
```diff
@@ -71,18 +71,23 @@ extern DeathHandler*          m_pDeathHandler;
...
+
 CPools* m_pools = 0;
 
 #define VAR_CWorld_IncludeCarTyres 0xb7cd70 // Used for CWorld_ProcessLineOfSight
 
 void InitFireInstantHit_MidHooks(void);
 void InitFireSniper_MidHooks(void);
 
+void HOOK_PedTakeStep(void);
```
Comment (botder) [inline Client/multiplayer_sa/multiplayer_shotsync.cpp]: Prefix the function with `static` to keep it in the translation unit ("this .cpp file")

File: Client/sdk/game/CPed.h [hunk 6]
```diff
@@ -143,6 +143,12 @@ enum eMoveAnim
     MOVE_SKATE,
 };
 
+enum eLandedPedFoot
```
Comment (botder) [inline Client/sdk/game/CPed.h]: Not needed?

File: Client/mods/deathmatch/logic/CClientGame.cpp [hunk 7]
```diff
@@ -278,6 +278,7 @@ CClientGame::CClientGame(bool bLocalPlay)
     g_pMultiplayer->SetGameEntityRenderHandler(CClientGame::StaticGameEntityRenderHandler);
     g_pMultiplayer->SetFxSystemDestructionHandler(CClientGame::StaticFxSystemDestructionHandler);
     g_pMultiplayer->SetDrivebyAnimationHandler(CClientGame::StaticDrivebyAnimationHandler);
+    g_pMultiplayer->SetTakePedAStepHandler(CClientGame::StaticPedTakeAStepHandler);
```
Comment (qaisjp) [inline Client/mods/deathmatch/logic/CClientGame.cpp]: `SetPedStepHandler`

File: Client/mods/deathmatch/logic/CClientGame.cpp [hunk 8]
```diff
@@ -6870,3 +6878,11 @@ void CClientGame::RemoveAnimationAssociationFromMap(CAnimBlendAssociationSAInter
 {
     m_mapOfCustomAnimationAssociations.erase(pAnimAssociation);
 }
+
+void CClientGame::PedTakeAStepHandler(CPedSAInterface* pPedSA, bool bFoot)
```
Comment (qaisjp) [inline Client/mods/deathmatch/logic/CClientGame.cpp]: `PedStepHandler` etc

**Review by qaisjp — CHANGES_REQUESTED** [review]

**Comment by jushar** [comment]
Why do you install the hooks in `multiplayer_shotsync.cpp`?

**Review by qaisjp — CHANGES_REQUESTED** [review]
Please fix as per Jusonex's comment

**Pull request merged**